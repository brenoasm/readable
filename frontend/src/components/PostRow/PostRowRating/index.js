import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Ratings from '../../Ratings';

import { UP_VOTE, DOWN_VOTE } from '../../../utils/ratings';

const propTypes = {
  post: PropTypes.object,
  modifyCommentVoteValues: PropTypes.func
};

const defaultProps = {
  post: {
    voteScore: 0
  },
  modifyCommentVoteValues: () => {}
};

const StyledPostRowRating = styled.div`
  i {
    font-size: x-large;
    cursor: pointer;
  }
`;

const PostRowRating = ({ post, modifyPostVoteValues }) => (
  <StyledPostRowRating>
    <Ratings
      voteScore={post.voteScore}
      rateDown={() => modifyPostVoteValues(post, DOWN_VOTE)}
      rateUp={() => modifyPostVoteValues(post, UP_VOTE)}
    />
  </StyledPostRowRating>
);

PostRowRating.propTypes = propTypes;
PostRowRating.defaultProps = defaultProps;

export default PostRowRating;
