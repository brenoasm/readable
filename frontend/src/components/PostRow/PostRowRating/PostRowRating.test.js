/* eslint-disable */

import React from 'react';

import PostRowRating from './';

describe('PostRowRating', () => {
  it('matches the snapshot', () => {
    const initialProps = {
      post: {
        id: '6ni6ok3ym7mf1p33lnez',
        timestamp: 1468479767190,
        title: 'Learn Redux in 10 minutes!',
        body:
          'Just kidding. It takes more than 10 minutes to learn technology.',
        author: 'thingone',
        category: 'redux',
        voteScore: -5,
        deleted: false,
        commentCount: 0
      },
      modifyPostVoteValue: jest.fn()
    };

    const wrapper = shallow(<PostRowRating {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
