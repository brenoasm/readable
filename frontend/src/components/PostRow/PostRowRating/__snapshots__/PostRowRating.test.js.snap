// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostRowRating matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostRowRating
    modifyCommentVoteValues={[Function]}
    modifyPostVoteValue={[Function]}
    post={
        Object {
            "author": "thingone",
            "body": "Just kidding. It takes more than 10 minutes to learn technology.",
            "category": "redux",
            "commentCount": 0,
            "deleted": false,
            "id": "6ni6ok3ym7mf1p33lnez",
            "timestamp": 1468479767190,
            "title": "Learn Redux in 10 minutes!",
            "voteScore": -5,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Ratings
        rateDown={[Function]}
        rateUp={[Function]}
        voteScore={-5}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "rateDown": [Function],
        "rateUp": [Function],
        "voteScore": -5,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Ratings
          rateDown={[Function]}
          rateUp={[Function]}
          voteScore={-5}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "rateDown": [Function],
          "rateUp": [Function],
          "voteScore": -5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
